src/test/kotlin/uk/gov/dluhc/printapi/mapper/PrintDetailsMapperTest.kt

```diff
@@ -0,0 +1,135 @@
+        given(sourceTypeMapper.toSourceTypeEntity(any())).willReturn(SourceTypeEntity.VOTER_CARD)
+
+        val expected = with(message) {
+            PrintDetails(
+                id = UUID.randomUUID(),
+                requestId = getAMongoDbId(),
+                sourceReference = sourceReference,
+                applicationReference = applicationReference,
+                sourceType = SourceTypeEntity.VOTER_CARD,
+                vacNumber = randomAlphanumeric(20),
+                requestDateTime = requestDateTime,
+                firstName = firstName,
+                middleNames = middleNames,
+                surname = surname,
+                certificateLanguage = certificateLanguageEntity,
+                photoLocation = photoLocation,
+                delivery = with(delivery) {
+                    CertificateDelivery(
+                        addressee = addressee,
+                        address = with(address) {
+                            Address(
+                                street = street,
+                                postcode = postcode,
+                                property = property,
+                                locality = locality,
+                                town = town,
+                                area = area,
+                                uprn = uprn
+                            )
+                        },
+                        deliveryClass = DeliveryClass.STANDARD
+                    )
+                },
+                gssCode = gssCode,
+                issuingAuthority = localAuthority.name,
+                issueDate = LocalDate.now(),
+                eroEnglish = with(ero) {
+                    ElectoralRegistrationOffice(
+                        name = name,
+                        phoneNumber = null,
+                        emailAddress = null,
+                        website = null,
+                        address = null
+                    )
+                },
+                eroWelsh = if (certificateLanguageModel == CertificateLanguage.EN) null else with(ero) {
+                    ElectoralRegistrationOffice(
+                        name = name,
+                        phoneNumber = null,
+                        emailAddress = null,
+                        website = null,
+                        address = null
+                    )
+                }
+            )
+        }
+        // When
+
+        val actual = mapper.toPrintDetails(message, ero, localAuthority.name)
+
+        // Then
+        assertThat(actual).usingRecursiveComparison().ignoringFields("id", "requestId", "vacNumber").isEqualTo(expected)
+        assertThat(actual.id).isNotNull
+        assertThat(actual.requestId).isNotNull.containsPattern(Regex("^[a-f\\d]{24}$").toPattern())
+        assertThat(actual.vacNumber).isNotNull.containsPattern(Regex("^[A-Za-z\\d]{20}$").toPattern())
```

Comment:
This assertion is fine for now, but _may_ have to change in the near future (just FYI)
We have a proposed VAC format (still waiting to get agreement on it) where we will use a 5 bit character set (to give us a total of 32 different chars in the char set), which in practice will mean 0-9 and A-Z, but excluding B, I, O and S (because they look like numbers)